### ISSUE TITLE
Duplicated identifiers in generated code

### ISSUE THREAD

**Issue by gneuvill** [derive4j/derive4j-fj#2]
Given
```java
public abstract class Property<T> implements __<Property.µ, T> {
    public enum µ {}
    Property() {}

    interface Cases<T, R> {
        R Simple(Class<T> ownerClass
            , String name
            , Value value
            , List<Param> params);

        R Multi(Class<T> ownerClass
            , String name
            , List<Value> values
            , List<Param> params);
    }
    public abstract <R> R match(Cases<T, R> cases);
}
```
we obtain
```java
public static <T> Ord<Property<T>> propertyOrd() {
    Ord<Property<T>> _propertyOrd = propertyOrd;
    if (_propertyOrd == null) {
      Ord<List<Property.Param>> listOrd = Ord.listOrd(_PropParam.paramOrd());
      Ord<List<Property.Value>> listOrd = Ord.listOrd(_PropVal.valueOrd());
      propertyOrd = _propertyOrd = Ord.ordDef(property1 -> property1.match(cases(
...
```
which doesn't compile (duplicate identifier `listOrd`)

**Comment by gneuvill** [comment]
Maybe I should have opened that in `derive4j` instead ?

**Comment by jbgi** [comment]
Good find! yeah, the bug is in derive4j. But since this plugin is probably the only user of the derive4j api for now, we can leave it here. Should be an easy fix.

**Comment by jbgi** [comment]
Should be fixed with derive4j  0.12.2.
Thanks for testing this out!
